### DISCUSSION TITLE
LPS Configured and CrazyFlie and I get the initial position in the air.

### DISCUSSION THREAD

**Discussion by danielcmdeveloper** [bitcraze/discussions#1586] [General]
I have installed and assembled LPS to establish a reference system for the CrazyFlie.
The coordinates of each of the anchors are shown in one of the images. The problem I am having is the following: Both the X and Y positions are being identified correctly but the Z axis has a strange behavior. In this case, it identifies that the drone is at Z = 0.38 from the ground, when the CF is on the ground without taking off.

Can someone help me with this?
Thank you very much 
![ConfiguredPositionPosition](https://github.com/user-attachments/assets/0df008c9-85ce-4a87-8c3f-8098cba349fe)
![DroneInAIR](https://github.com/user-attachments/assets/51998475-3d05-435d-8407-955380c0141b)

**Comment by ToveRumar** [comment]
Hi @danielcmdeveloper!

Looking at this issue its unfortunately an inherent trait of the LPS system while taking of from the floor. 

The z-position is more prone to  offset errors  due to the angle to the antennas being steep and the signals being more interfered/reflected closer to the ground.  You could mitigate this by trying to have the floor being as empty as possible and not using reflective materials close to the system. 
A z-ranger could also help you with this issue. Otherwise you need to programatically take this into account. 

If you are using the cflib you need to set the first setpoint to something that is higher than 0.38 and it will then correct it self as the drone moves up. 

I hope this answer helps you understand more about what is happening.

**Reply by danielcmdeveloper** [reply]
Good morning,

Perfect, thank you very much for your answer, I will implement the corrections that you suggest and I will come back to give you a feedback of the result.
Taking advantage of this thread, the positional offset is not only in the Z axis but also in the X and Y axes. Is this also normal?

Thank you very much.
Best regards
Dani

**Reply by ToveRumar** [reply]
Hi!

Great, please do that!
It would help us document this for other users as well. 

It depends a bit on how much of an offset you see. The LPS system has about 10cm accuracy so if the offset is in this area it could be normal. The accuracy depends also on the placement of the LPS nodes. 
If you have not seen this recommendation for setting up the nodes, this could be good information;
https://www.bitcraze.io/documentation/tutorials/getting-started-with-loco-positioning-system/#place-the-anchors-in-the-room

I hope this helps in setting up the LPS system!

**Reply by danielcmdeveloper** [reply]
Good morning,
After some research and thanks to your help, we have repositioned the anchors with the right dimensions. It is true that there is still a positioning error but we have solved it with an offset to compensate that error. We do it in real time, since we are developing with the Python API of CrazyFlie (cflib). In this case, by performing the compensation with the offset we almost get the position in which the drone is located.

Thank you very much for the help.
Regards

**Reply by ToveRumar** [reply]
Hi!

Ok, great! I will consider updating the documentation around LPS so that this is clearer. 
Thank you for coming back with your results!

**Comment by danielcmdeveloper** [comment]
Thank you very much for your help.
Regards
